src/main/java/us/warframestat/moddetection/api/API.java

```diff
@@ -0,0 +1,69 @@
+package us.warframestat.moddetection.api;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Entrypoint on the API.
+ */
+public class API {
+    public static final Logger LOGGER = LoggerFactory.getLogger(API.class);
+
+    public static File DATA = new File(System.getProperty("java.io.tmpdir"), "ModDetectionOpenCV");
+
+    /**
+     * Run the detection.
+     * @param args datafolder
+     */
+    public static void main(String[] args) {
+        if (args.length != 0) {
+            try {
+                DATA = new File(args[0]);
+                return;
+            } catch (NullPointerException e) {
+                DATA = new File(System.getProperty("java.io.tmpdir"), "ModDetectionOpenCV");
+            }
+        }
+        try {
+            API.setupData();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Setup the temp data directory.
+     * @throws IOException  if the directory cannot be created.
+     */
+    public static void setupData() throws IOException {
+        DATA.mkdirs();
+        FileUtils.cleanDirectory(DATA);
+        File cascade = DATA.toPath().resolve("cascade").toFile();
+        File tesseract = DATA.toPath().resolve("tesseract").toFile();
```

Comment:
We probably need to do this in startup on main to make sure we don't crash mid-process